orderedmap.go

```diff
@@ -6,15 +6,28 @@ type orderedMapElement struct {
 	key, value interface{}
 }
 
+// OrderedMap the ordered map data structure
 type OrderedMap struct {
-	kv map[interface{}]*list.Element
-	ll *list.List
+	kv      map[interface{}]*list.Element
+	ll      *list.List
+	maxSize int
```

Comment:
I don't want to pollute `OrderedMap` with features in this way. Instead you can create a new type that encapsulates the `OrderedMap`.